### PR TITLE
Fix cluster name used during upgrade for CAPI objects

### PR DESCRIPTION
**Pull request by mrajashree**
This commit includes two changes:
* Post upgrade eks-a waits for ControlPlaneReady and ManagedEtcdReady conditions
on the cluster. This check in the UpgradeCluster method in cluster_manager.go is
done on the workload cluster. In case of single management cluster, the workload
cluster in this method ends up being the management cluster itself since that maintains
all the CAPI objects. So this check will happen always only for the management cluster
instead of the target cluster undergoing upgrade.
This commit fixes this by passing the target workload cluster's name to the methods
that wait for the conditions post-upgrade
* Passes in the exact target cluster's name to retrieve the EtcdadmCluster CR
from the management cluster. This is how the KCP and MD objects are retrieved too

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/545#" title="Author self-approved">mrajashree</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [mrajashree]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by abhay-krishna** [comment]
@mrajashree: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-e2e-presubmit | f7bfb9532e3dcd5f15b61773ed5c5d73ef712f52 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/545/eks-anywhere-e2e-presubmit/1455331541541457920) | `/test eks-anywhere-e2e-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=545). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:mrajashree).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by abhay-krishna** [comment]
@mrajashree: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-e2e-presubmit | f7bfb9532e3dcd5f15b61773ed5c5d73ef712f52 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/545/eks-anywhere-e2e-presubmit/1455331541541457920) | `/test eks-anywhere-e2e-presubmit`
eks-anywhere-attribution-files-presubmit | f7bfb9532e3dcd5f15b61773ed5c5d73ef712f52 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/545/eks-anywhere-attribution-files-presubmit/1455331541503709184) | `/test eks-anywhere-attribution-files-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=545). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:mrajashree).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by abhay-krishna** [comment]
@mrajashree: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-e2e-presubmit | f7bfb9532e3dcd5f15b61773ed5c5d73ef712f52 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/545/eks-anywhere-e2e-presubmit/1455331541541457920) | `/test eks-anywhere-e2e-presubmit`
eks-anywhere-attribution-files-presubmit | f7bfb9532e3dcd5f15b61773ed5c5d73ef712f52 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/545/eks-anywhere-attribution-files-presubmit/1455331541503709184) | `/test eks-anywhere-attribution-files-presubmit`
eks-anywhere-mocks-presubmits | f7bfb9532e3dcd5f15b61773ed5c5d73ef712f52 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/545/eks-anywhere-mocks-presubmits/1455331541587595264) | `/test eks-anywhere-mocks-presubmits`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=545). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:mrajashree).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by abhay-krishna** [comment]
@mrajashree: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-e2e-presubmit | f7bfb9532e3dcd5f15b61773ed5c5d73ef712f52 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/545/eks-anywhere-e2e-presubmit/1455331541541457920) | `/test eks-anywhere-e2e-presubmit`
eks-anywhere-attribution-files-presubmit | f7bfb9532e3dcd5f15b61773ed5c5d73ef712f52 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/545/eks-anywhere-attribution-files-presubmit/1455331541503709184) | `/test eks-anywhere-attribution-files-presubmit`
eks-anywhere-mocks-presubmits | f7bfb9532e3dcd5f15b61773ed5c5d73ef712f52 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/545/eks-anywhere-mocks-presubmits/1455331541587595264) | `/test eks-anywhere-mocks-presubmits`
eks-anywhere-presubmit | f7bfb9532e3dcd5f15b61773ed5c5d73ef712f52 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/545/eks-anywhere-presubmit/1455331541637926912) | `/test eks-anywhere-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=545). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:mrajashree).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by abhay-krishna** [comment]
@mrajashree: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-e2e-presubmit | f7bfb9532e3dcd5f15b61773ed5c5d73ef712f52 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/545/eks-anywhere-e2e-presubmit/1455331541541457920) | `/test eks-anywhere-e2e-presubmit`
eks-anywhere-presubmit | f568103f1cec969bd0ca9ea834a14df1701fca0c | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/545/eks-anywhere-presubmit/1455336692008357888) | `/test eks-anywhere-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=545). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:mrajashree).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Review by mitalipaygude — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/545#pullrequestreview-794805895" title="Approved">mitalipaygude</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/545#" title="Author self-approved">mrajashree</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [mitalipaygude,mrajashree]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
